apps/avatax/src/modules/calculate-taxes/use-case/calculate-taxes.use-case.ts

```diff
@@ -1,18 +1,30 @@
+import { AuthData } from "@saleor/app-sdk/APL";
 import { createLogger } from "@saleor/apps-logger";
+import * as Sentry from "@sentry/nextjs";
+import { captureException } from "@sentry/nextjs";
+import { waitUntil } from "@vercel/functions";
+import { Result, err, fromPromise } from "neverthrow";
+import { MetadataItem } from "../../../../generated/graphql";
 import { BaseError } from "../../../error";
 import { AppConfigExtractor, IAppConfigExtractor } from "../../../lib/app-config-extractor";
-import { CalculateTaxesPayload } from "../../webhooks/payloads/calculate-taxes-payload";
-import { AuthData } from "@saleor/app-sdk/APL";
-import { verifyCalculateTaxesPayload } from "../../webhooks/validate-webhook-payload";
-import { TaxIncompletePayloadErrors } from "../../taxes/tax-error";
-import { err, fromPromise, Result } from "neverthrow";
 import { AppConfigurationLogger } from "../../../lib/app-configuration-logger";
-import * as Sentry from "@sentry/nextjs";
-import { captureException } from "@sentry/nextjs";
 import { AvataxCalculateTaxesResponse } from "../../avatax/calculate-taxes/avatax-calculate-taxes-adapter";
-import { MetadataItem } from "../../../../generated/graphql";
+import {
+  TaxesCalculatedInCheckoutLog,
+  TaxesCalculatedInOrderLog,
+  TaxesCalculationFailedConfigErrorLog,
+  TaxesCalculationFailedInvalidPayloadLog,
+  TaxesCalculationFailedUnhandledErrorLog,
+  TaxesCalculationProviderErrorLog,
+} from "../../public-log-drain/public-events";
+import { PublicLogDrain } from "../../public-log-drain/public-log-drain";
+import { LogDrainJsonTransporter } from "../../public-log-drain/transporters/public-log-drain-json-transporter";
+import { LogDrainOtelTransporter } from "../../public-log-drain/transporters/public-log-drain-otel-transporter";
+import { TaxIncompletePayloadErrors } from "../../taxes/tax-error";
+import { CalculateTaxesPayload } from "../../webhooks/payloads/calculate-taxes-payload";
+import { verifyCalculateTaxesPayload } from "../../webhooks/validate-webhook-payload";
 
-export class CalculateTaxesUseCase {
+class PrivateCalculateTaxesUseCase {
```

Comment:
I'd rethink this. Technically extending works, but I think we can make it simpler. Maybe keep single usecase class ("final") and consider one of these:
1. making `private execute()` that will be implementing current "calculateTaxes" internals and "calculateTaxes" will only run "this.execute" and mapp errors
2. Add decorator on method to wrap errors
3. Add "errorAndEmit" private method that will receive error, emit and return